### PR TITLE
rocksdb: Check return value of DecodeHLCTimestamp

### PR DESCRIPTION
**Pull request by a-robinson**
Probably insignificant, but it seems like bad hygiene to be ignoring these return values.

@danhhz @petermattis

---
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/13099)

### CODE REVIEW

File: pkg/storage/engine/rocksdb/db.cc [hunk 1]
```diff
@@ -1404,12 +1404,22 @@ DBString DBEngine::GetUserProperties() {
 
     auto ts_min = userprops.find("crdb.ts.min");
     if (ts_min != userprops.end()) {
-      DecodeHLCTimestamp(rocksdb::Slice(ts_min->second), sst->mutable_ts_min());
+      if (!DecodeHLCTimestamp(rocksdb::Slice(ts_min->second), sst->mutable_ts_min())) {
+        google::protobuf::SStringPrintf(all.mutable_error(),
+              "unable to decode crdb.ts.min value '%s' in table %s",
+              ts_min->second.c_str(), sst->path().c_str());
+        return ToDBString(all.SerializeAsString());
```
Comment (petermattis) [inline pkg/storage/engine/rocksdb/db.cc]: `break`?

File: pkg/storage/engine/rocksdb/db.cc [hunk 2]
```diff
@@ -1404,12 +1404,22 @@ DBString DBEngine::GetUserProperties() {
 
     auto ts_min = userprops.find("crdb.ts.min");
     if (ts_min != userprops.end()) {
-      DecodeHLCTimestamp(rocksdb::Slice(ts_min->second), sst->mutable_ts_min());
+      if (!DecodeHLCTimestamp(rocksdb::Slice(ts_min->second), sst->mutable_ts_min())) {
+        google::protobuf::SStringPrintf(all.mutable_error(),
+              "unable to decode crdb.ts.min value '%s' in table %s",
+              ts_min->second.c_str(), sst->path().c_str());
```
Comment (petermattis) [inline pkg/storage/engine/rocksdb/db.cc]: Maybe format the timestamp as hex. I think something like `rocksdb::Slice(ts_min->second).ToString(true).c_str()`.

**Review by petermattis — APPROVED** [review]
LGTM

**Comment by bdarnell** [comment]
LGTM

Maybe we should add the `warn_unused_result` attribute to this function.

File: pkg/storage/engine/rocksdb/db.cc (hunk 2 above)
Comment (a-robinson) [inline pkg/storage/engine/rocksdb/db.cc]: Sure, done.

File: pkg/storage/engine/rocksdb/db.cc (hunk 1 above)
Comment (a-robinson) [inline pkg/storage/engine/rocksdb/db.cc]: Done.

**Review by a-robinson — COMMENTED** [review]
@danhhz - is there a potential issue here given that we appear to blindly return both the min and max properties in `TimeBoundTblPropCollector::Finish` regardless of whether they're populated? Should that be changed to be conditional on them being non-empty?

**Comment by a-robinson** [comment]
I see. Added a check to allow for empty timestamps in the (presumably) rare case of empty sstables.

**Comment by a-robinson** [comment]
@bdarnell - it doesn't look like we use the `warn_unused_result` attribute at all. I'm not going to introduce it right now, but perhaps it deserves an issue?

**Comment by bdarnell** [comment]
Filed #13101

**Pull request merged**